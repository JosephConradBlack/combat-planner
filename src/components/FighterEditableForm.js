import React, { Component } from 'react';
import styled from 'styled-components';

import {
  Segment,
  Header,
  Icon,
  Form,
  Checkbox,
  Responsive,
  Divider,
  Button
} from 'semantic-ui-react';

const StyledIcon = styled(Icon)`
  &&& {
    margin-top: 26px;
  }
`;

class FighterEditableForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      initiative: '',
      health: '',
      armor: '',
      player: false
    };
  }

  handleChange = (e, target) => {
    const { name, value, checked, type } = target ? target : e.target;

    if (type && type === 'checkbox') {
      this.setState(Object.assign({}, this.state, { [name]: checked }));
    } else {
      this.setState(Object.assign({}, this.state, { [name]: value }));
    }
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.onSubmit(this.state);
  };

  handleCancel = e => {
    e.preventDefault();

    this.setState({
      name: '',
      initiative: '',
      health: '',
      armor: '',
      player: ''
    });
  };

  render() {
    return (
      <Segment>
        <Form>
          <Header as="h2">Add fighter</Header>
          <Form.Input
            required
            label="Fighter Name"
            placeholder="Goblin #1"
            name="name"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <Form.Group widths="equal">
            <Form.Field
              type="number"
              control="input"
              width={4}
              label="Initiative"
              placeholder="17"
              name="initiative"
              value={this.state.initiative}
              onChange={this.handleChange}
            />

            <Responsive
              as={StyledIcon}
              minWidth={768}
              size="big"
              name="heart outline"
              color="red"
            />
            <Form.Field
              type="number"
              control="input"
              label="Health"
              placeholder="30"
              name="health"
              value={this.state.health}
              onChange={this.handleChange}
            />

            <Responsive
              as={StyledIcon}
              minWidth={768}
              size="big"
              name="shield"
              color="grey"
            />
            <Form.Field
              type="number"
              control="input"
              label="Armor"
              placeholder="18"
              name="armor"
              onChange={this.handleChange}
            />
          </Form.Group>
          <Checkbox
            label="Is this a player?"
            name="player"
            checked={this.state.player}
            onChange={this.handleChange}
          />
          <Divider />
          <Button basic content="Cancel" onClick={this.handleCancel} />
          <Button positive floated="right" content="Add" />
        </Form>
      </Segment>
    );
  }
}

FighterEditableForm.propTypes = {};

export default FighterEditableForm;
