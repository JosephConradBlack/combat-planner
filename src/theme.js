export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
