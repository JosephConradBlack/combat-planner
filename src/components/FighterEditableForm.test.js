import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import FighterEditableForm from './FighterEditableForm';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('FighterEditable', () => {
  it('renders with all required props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <FighterEditableForm onSubmit={() => {}} />
      </ThemeProvider>
    ).dive();

    expect(toJson(component)).toMatchSnapshot();
  });
});
