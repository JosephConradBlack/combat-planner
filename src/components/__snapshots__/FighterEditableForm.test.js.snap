// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FighterEditable renders with no props 1`] = `
<div
  className="ui segment"
>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <h2
      className="ui header"
    >
      Add fighter
    </h2>
    <div
      className="required field"
    >
      <label
        htmlFor={undefined}
      >
        Fighter Name
      </label>
      <div
        className="ui input"
        content={undefined}
      >
        <input
          disabled={undefined}
          name="name"
          onChange={[Function]}
          placeholder="Goblin #1"
          required={true}
          tabIndex={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="equal width fields"
    >
      <div
        className="four wide field"
      >
        <label
          htmlFor={undefined}
        >
          Initiative
        </label>
        <input
          content={undefined}
          disabled={undefined}
          name="initiative"
          onChange={[Function]}
          placeholder="17"
          required={undefined}
          type="number"
          value=""
        />
      </div>
      <i
        aria-hidden="true"
        className="red heart outline big icon sc-bdVaJa iAKyKJ"
      />
      <div
        className="field"
      >
        <label
          htmlFor={undefined}
        >
          Health
        </label>
        <input
          content={undefined}
          disabled={undefined}
          name="health"
          onChange={[Function]}
          placeholder="30"
          required={undefined}
          type="number"
          value=""
        />
      </div>
      <i
        aria-hidden="true"
        className="grey shield big icon sc-bdVaJa iAKyKJ"
      />
      <div
        className="field"
      >
        <label
          htmlFor={undefined}
        >
          Armor
        </label>
        <input
          content={undefined}
          disabled={undefined}
          name="armor"
          onChange={[Function]}
          placeholder="18"
          required={undefined}
          type="number"
        />
      </div>
    </div>
    <div
      className="ui checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        disabled={undefined}
        id={undefined}
        name="player"
        onClick={[Function]}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value={undefined}
      />
      <label
        htmlFor={undefined}
      >
        Is this a player?
      </label>
    </div>
    <div
      className="ui divider"
    />
    <button
      className="ui basic button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Cancel
    </button>
    <button
      className="ui positive right floated button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Add
    </button>
  </form>
</div>
`;
